node/web/controllers/authController.js

```diff
@@ -8,40 +8,69 @@ module.exports.factory = function (router, repo) {
         addCookie;
 
     addCookie = function (user, res) {
-        // normally, you wouldn't set a plain old user object as the
-        // value of the cookie - it's not secure.
         res.cookie('auth', user, { maxAge: maxAge, httpOnly: true });
     };
 
-
     router.post('/register', function (req, res) {
-        repo.create(req.body, function (err, result) {
-            if (!err && result.insertedId) {
-                repo.get(req.body.email, function (err, user) {
-                    if (!err) {
-                        addCookie(user, res);
-                        res.redirect('/');
-                    } else {
-                        res.status(400);
-                    }
-                });
-            } else {
-                res.status(400);
-            }
-        });
+		if (req.body.email == "" || req.body.name == "" ||
+			req.body.userId == "") {
+			res.redirect('/registerError');
+			return;
+		}
+
+		repo.create(req.body, function (err, result) {
+			if (!err && result.insertedId) {
+				var doc = {
+					email: req.body.email,
+					userId: req.body.userId,
+					name: req.body.name
+				}
+				addCookie(doc, res);
+				res.redirect('/index');
+				return;
+			} else {
+				if (err.message.indexOf("unq.users.email") >= 0) {
+					res.redirect('/registerError');
+					return;
+				} else if (err.message.indexOf("unq.users.id") >= 0) {
+					res.redirect('/registerError');
+					return;
+				} else {
+					res.redirect('/registerError');
+					return;
+				}
+				return;
```

Comment:
the nested `if` block is unnecessary, you could just use `else if` in the outer if block instead.